manifests/init.pp

```diff
@@ -376,6 +376,10 @@
     if $agent_major_version != undef and $agent_major_version != $_agent_major_version {
       fail('Provided and deduced agent_major_version don\'t match')
     }
+    if $facts['os']['name'] == 'CentOS' and $facts['os']['name']['distro']['release']['major'] < 7 and 0 + $3 > 51 {
+      warn('datadog-agent '+ $agent_major_version + '.51 is the last supported version on CentOS 6. Installing ' + $agent_major_version + '.51 now')
+      $agent_version='1:7.51.1'
```

Comment:
Shouldn't we be erroring out in this case? That's what we've been doing on the rest of install methods when the user pins the version.